tests/components/script/test_init.py

```diff
@@ -679,6 +679,7 @@ async def test_script_variables(hass, caplog):
             "script": {
                 "script1": {
                     "variables": {
+                        "this_var": "{{this.entity_id}}",
```

Comment:
```suggestion
                        # Verify the this variable is available when rendering variables
                        "this_var": "{{this.entity_id}}",
```

Reply:
Restructured to test for this during rendering of variables, and rendering of templates in actions.